db/model/sample.js

```diff
@@ -254,16 +255,32 @@ module.exports = function sample(seq, dataTypes) {
         const curr = now || new Date();
         let numberTimedOut = 0;
         let numberEvaluated = 0;
+        const timedOutSamples = [];
         return new seq.Promise((resolve, reject) => {
           Sample.scope('checkTimeout').findAll()
           .each((s) => {
             numberEvaluated++;
             if (s.aspect && u.isTimedOut(s.aspect.timeout, curr, s.updatedAt)) {
               numberTimedOut++;
+
+              /*
+               * NOTE: a separate copy of the sample needed to be made (by
+               * deep cloning the required fields) to let the garbage collector
+               * cleanup the timedOutSamples array.
+               */
+              const timedOutSample = {
+                value: constants.statuses.Timeout,
+                status: constants.statuses.Timeout,
+                name: s.name.split()[0],
```

Comment:
Yes, it creates a copy of the string instead. There is a note about it a few lines above.